satpy/readers/fci_l2_nc.py

```diff
@@ -0,0 +1,247 @@
+                creation_time: creation time of the product
+                platform_name: name of the platform
+
+        """
+        attributes = {
+            'filename': self.filename,
+            'start_time': self.start_time,
+            'end_time': self.end_time,
+            'spacecraft_name': self.spacecraft_name,
+            'ssp_lon': self.ssp_lon,
+            'sensor': self.sensor,
+            'creation_time': self.filename_info['creation_time'],
+            'platform_name': self.spacecraft_name,
+        }
+        return attributes
+
+    def get_area_def(self, key):
+        """Return the area definition (common to all data in product)."""
+        return self._area_def
+
+    def _compute_area_def(self):
+        """Compute the area definition.
+
+        Returns:
+            AreaDefinition: A pyresample AreaDefinition object containing the area definition.
+
+        """
+        # Read the projection data from the mtg_geos_projection variable
+        a = float(self._projection.attrs['semi_major_axis'])
+        b = float(self._projection.attrs['semi_minor_axis'])
+        h = float(self._projection.attrs['perspective_point_height'])
+
+        # TODO sweep_angle_axis value not handled at the moment, therefore commented out
+        # sweep_axis = self._projection.attrs['sweep_angle_axis']
+
+        # Coordinates of the pixel in radians
+        x = self.nc['x']
+        y = self.nc['y']
+        # TODO conversion to radians: offset and scale factor are missing from some test NetCDF file
+        # TODO the next two lines should be removed when the offset and scale factor are correctly configured
+        if not hasattr(x, 'standard_name'):
+            x = np.radians(x * 0.003202134 - 8.914740401)
+            y = np.radians(y * 0.003202134 - 8.914740401)
+
+        # Convert to degrees as required by the make_ext function
+        x_deg = np.degrees(x)
+        y_deg = np.degrees(y)
+
+        # Select the extreme points of the extension area
+        x_l, x_r = x_deg.values[0], x_deg.values[-1]
+        y_l, y_u = y_deg.values[0], y_deg.values[-1]
+
+        # Compute the extension area in meters
+        area_extent = make_ext(x_l, x_r, y_l, y_u, h)
+
+        # Assemble the projection definition dictionary
+        p_dict = {
+            'nlines': self.nlines,
+            'ncols': self.ncols,
+            'ssp_lon': self.ssp_lon,
+            'a': a,
+            'b': b,
+            'h': h,
+            'a_name': 'FCI Area',                 # TODO to be confirmed
+            'a_desc': 'Area for FCI instrument',  # TODO to be confirmed
+            'p_id': 'geos'
+        }
+
+        # Compute the area definition
+        area_def = get_area_definition(p_dict, area_extent)
+
+        return area_def
+
+    def __del__(self):
+        """Close the NetCDF file that may still be open."""
+        try:
+            self.nc.close()
+        except AttributeError:
+            pass
```

Comment:
This would be taken care of it the `NetCDF4FileHandler` was used as base class. Also, important optimisations have been implemented in that class, so I would recommend to consider using it.

Reply:
@djhoese says on slack that this is not necessarily the case. A quick benchmarking would be nice :)